src/main/clojure/clara/rules/compiler.clj

```diff
@@ -1576,6 +1534,60 @@
             (swap! alpha-map assoc fact-type new-nodes)
             [new-nodes facts]))))))
 
+(sc/defn build-network
+  "Constructs the network from compiled beta tree and condition functions."
+  [id-to-node :- {sc/Int sc/Any}
+   beta-roots
+   alpha-fns
+   productions
+   fact-type-fn
+   ancestors-fn
+   activation-group-sort-fn
+   activation-group-fn]
+
+  (let [beta-nodes (vals id-to-node)
+
+        production-nodes (for [node beta-nodes
+                               :when (= ProductionNode (type node))]
+                           node)
+
+        query-nodes (for [node beta-nodes
+                          :when (= QueryNode (type node))]
+                      node)
+
+        query-map (into {} (for [query-node query-nodes
+
+                                 ;; Queries can be looked up by reference or by name;
+                                 entry [[(:query query-node) query-node]
+                                        [(:name (:query query-node)) query-node]]]
+                             entry))
+
+        ;; type, alpha node tuples.
+        alpha-nodes (for [{:keys [type alpha-fn children env] :as alpha-map} alpha-fns
+                          :let [beta-children (map id-to-node children)]]
+                      [type (with-meta
+                              (eng/->AlphaNode env beta-children alpha-fn)
+                              {:alpha-expr (:alpha-expr (meta alpha-map))})])
+
+        ;; Merge the alpha nodes into a multi-map
+        alpha-map (reduce
+                   (fn [alpha-map [type alpha-node]]
+                     (update-in alpha-map [type] conj alpha-node))
+                   {}
+                   alpha-nodes)
+
+        get-alphas-fn (create-get-alphas-fn fact-type-fn ancestors-fn alpha-map)]
+
+    (strict-map->Rulebase
+     {:alpha-roots alpha-map
+      :beta-roots beta-roots
+      :productions productions
+      :production-nodes production-nodes
+      :query-nodes query-map
+      :id-to-node id-to-node
+      :get-alphas-fn get-alphas-fn
```

Comment:
It doesn't really matter, but it'd make more sense to me to keep these keys in the same order as the basis fields of the Rulebase record.